hints/hint_distribution.py

```diff
@@ -189,28 +193,20 @@ def start(self, logic: Logic, always_hints: list, sometimes_hints: list):
         self.hinted_locations.extend(self.logic.prerandomization_item_locations.keys())
 
         # populate our internal list copies for later manipulation
-        for sots_loc, item in self.logic.rando.sots_locations.items():
-            if item in self.removed_items:
-                continue
-            if self.logic.rando.options["small-key-mode"] not in [
-                "Anywhere",
-                "Lanayru Caves Key Only",
-            ]:
-                # don't hint small keys unless keysanity is on
-                if item.endswith("Small Key"):
-                    continue
-            elif self.logic.rando.options["small-key-mode"] == "Lanayru Caves Key Only":
-                if item.endswith("Small Key") and item != "LanayruCaves Small Key":
-                    continue
-
-            if self.logic.rando.options["boss-key-mode"] not in ["Anywhere"]:
-                # don't hint boss keys unless keysanity is on
-                if item.endswith("Boss Key"):
-                    continue
-
-            zone, specific_loc = Logic.split_location_name_by_zone(sots_loc)
-            self.sots_locations.append((zone, sots_loc, item))
+        self.sots_locations = self.loc_dict_filter(self.logic.sots_locations)
         self.rng.shuffle(self.sots_locations)
+        self.goals = list(self.logic.goal_locations.keys())
+        self.rng.shuffle(
+            self.goals
+        )  # shuffle the goal names that will be chosen in sequence when goal hints are placed to try to ensure one is placed for each goal
+        self.goal_locations = list(
+            [
+                (self.loc_dict_filter(self.logic.goal_locations[goal_name]))
+                for goal_name in self.goals
+            ]
+        )  # create corresponding list of shuffled goal items
```

Comment:
```suggestion
        # create corresponding list of shuffled goal items
        self.goal_locations = [
            (self.loc_dict_filter(self.logic.goal_locations[goal_name]))
            for goal_name in self.goals
        ]
```
list([...]) is redundant, only [...] works the same way